core/time/rfc3339.odin

```diff
@@ -187,4 +187,108 @@ scan_digits :: proc(s: string, sep: string, count: int) -> (res: int, ok: bool)
 		found_sep |= rune(s[count]) == v
 	}
 	return res, found_sep
-}
\ No newline at end of file
+}
+
+/*
+Serialize the timestamp as a RFC 3339 string.
+
+The boolean `ok` is false if the `time` is not a valid datetime, or if allocating the result string fails.
+
+**Inputs**:
+- `utc_offset`: offset in minutes wrt UTC (ie. the timezone)
+- `include_nanos`: whether to include nanoseconds in the result.
+*/
+time_to_rfc3339 :: proc(time: Time, utc_offset : int = 0, include_nanos := true, allocator := context.allocator) -> (res: string, ok: bool) {
+	// convert to datetime
+	datetime := time_to_datetime(time) or_return
+
+	if datetime.year < 0 || datetime.year >= 10_000 { return "", false }
+
+	temp_string := [30]u8{}
```

Comment:
```suggestion
	temp_string := [36]u8{}
```

Date: 10 characters
`T`: 1 character
Time: 8 characters
`.`: 1 character
Nanos: 9 characters max
`Z`: 1 character
tz offset: 6 characters

36 characters maximum for an RFC3339 timestamp.